lib/beaker/tasks/quick_start.rb

```diff
@@ -1,49 +1,85 @@
-      fh.print(cli.execute)
+    if File.exist?(hosts_file) then
+      File.open(hosts_file, 'w') do |fh|
+        fh.print(cli.execute)
+      end
+    else
+      puts "Host Config File: #{hosts_file} already exists so not overwriting"
     end
   end
 
+
   desc 'Generate Default Pre-Suite'
   task :gen_pre_suite do
+    pre_suite_file = "acceptance/setup/default_pre_suite.rb"
     FileUtils.mkdir_p('acceptance/setup') # -p ignores when dir already exists
-    File.open("acceptance/setup/default_pre_suite.rb", 'w') do |fh|
-      fh.print('install_puppet')
+    if File.exist?(pre_suite_file) then
+      File.open(pre_suite_file, 'w') do |fh|
+        fh.print('install_puppet')
+      end
+    else
+      puts "Pre Suite File: #{pre_suite_file} already exists so not overwriting"
     end
   end
 
   desc 'Generate Default Smoke Test'
   task :gen_smoke_test do
+    smoke_test_file = "acceptance/setup/default_pre_suite.rb"
     FileUtils.mkdir_p('acceptance/tests') # -p ignores when dir already exists
-    File.open("acceptance/tests/default_smoke_test.rb", 'w') do |fh|
-      fh.print("test_name 'puppet install smoketest'
+    if File.exist?(smoke_test_file) then
+      File.open("acceptance/tests/default_smoke_test.rb", 'w') do |fh|
+        fh.print("test_name 'puppet install smoketest'
 step 'puppet install smoketest: verify \\'puppet help\\' can be successfully called on
 all hosts'
     hosts.each do |host|
       on host, puppet('help')
     end")
+      end
+    else
+      puts "Smoke Test File: #{smoke_test_file} already exists so not overwriting"
     end
   end
 
-  desc 'Run Default Smoke Test'
-  task :run => ['beaker_quickstart:gen_hosts', 'beaker_quickstart:gen_pre_suite', 'beaker_quickstart:gen_smoke_test'] do
-    system(beaker_command)
+  desc 'Run Default Smoke Test, after generating default host config and test files, valid options are: vmpooler or vagrant.'
+  task :run_test, [:provisioner] => ["beaker_quickstart:gen_hosts", 'beaker_quickstart:gen_pre_suite',
+                                     'beaker_quickstart:gen_smoke_test'] do
+  |t, args|
+    provisioner             = args[:provisioner] ||='vmpooler'
+    system_args             = Hash.new
+    system_args[:hosts]     = "acceptance/config/default_#{provisioner}_hosts.yaml"
+    system_args[:pre_suite] = 'acceptance/setup/default_pre_suite.rb'
+    system_args[:tests]     = 'acceptance/tests/default_smoke_test.rb'
+    system(beaker_command(system_args))
   end
 
 end
 
-def beaker_command
+
+def beaker_command(system_args)
+  binding.pry
   cmd_parts = []
   cmd_parts << "beaker"
-  cmd_parts << "--hosts acceptance/config/default_hosts.yaml"
-  cmd_parts << "--pre-suite acceptance/setup/default_pre_suite.rb"
-  cmd_parts << "--tests acceptance/tests/default_smoke_test.rb"
+  cmd_parts << "--hosts #{system_args[:hosts]}"
+  cmd_parts << "--pre-suite #{system_args[:pre_suite]}"
+  cmd_parts << "--tests #{system_args[:tests]}"
   cmd_parts.flatten.join(" ")
```

Comment:
should this beaker_command have an ENV setup so one can add arbitrary arguments?  rototiller will largely move us away from this, but i've found it useful in the past.  it would allow users to retain this Rakefile and expand functionality without edits.